### ISSUE TITLE
Unit test failure in TrackingRecHit2DCUDA_t

### ISSUE THREAD

**Issue by makortel** [cms-sw/cmssw#36605]
The unit test TrackingRecHit2DCUDA_t is failing in GPU_X IBs (also in PPC) since CMSSW_12_3_GPU_X_2021-12-23-2300
```
===== Test "TrackingRecHit2DCUDA_t" ====
TrackingRecHit2DCUDA_t: /pool/condor/dir_235681/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/c3d1c77324e90ad940179994e71ef3ef/opt/cmssw/slc7_amd64_gcc10/cms/cmssw-patch/CMSSW_12_3_GPU_X_2021-12-23-2300/src/CUDADataFormats/TrackingRecHit/test/TrackingRecHit2DCUDA_t.cpp:30: int main(): Assertion `tkhitH.view()->nMaxModules() == unsigned(nModules)' failed.
timeout: the monitored command dumped core
/bin/sh: line 1: 22064 Aborted                 timeout 3600 TrackingRecHit2DCUDA_t

---> test TrackingRecHit2DCUDA_t had ERRORS
```
https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc10/CMSSW_12_3_GPU_X_2021-12-23-2300/unitTestLogs/CUDADataFormats/TrackingRecHit#/

**Comment by makortel** [comment]
assign reconstruction, heterogeneous

FYI @cms-sw/trk-dpg-l2 @AdrianoDee

**Comment by makortel** [comment]
I guess the failure would be related to #36235 where the assertion is added, but I have no idea why the test is failing in the IBs but did not fail in the PR tests.

**Comment by mmusich** [comment]
I think this can be closed, as it was solved by #36607, right?

**Comment by makortel** [comment]
> I think this can be closed, as it was solved by #36607, right?

Yup.

**Comment by makortel** [comment]
+heterogeneous
